Exoft.Gamification.Api.Test/AchievementServiceTests.cs

```diff
@@ -0,0 +1,151 @@
+                _mapper, _unitOfWork.Object);
+        }
+
+        [TestCaseSource(typeof(TestCaseSources), nameof(TestCaseSources.ValidCreateAchievementModels))]
+        public async Task AddAchievementAsync_CreateAchievementModel_ReturnsReadAchievementModel(CreateAchievementModel model)
+        {
+            //Arrange
+            var expectedValue = AchievementDumbData.GetReadAchievementModel(model);
+
+            _fileService.Setup(x => x.AddOrUpdateFileByIdAsync(It.IsAny<IFormFile>(), It.IsAny<Guid?>())).Returns(Task.FromResult(expectedValue.IconId));
+            _achievementRepository.Setup(x => x.AddAsync(It.IsAny<Achievement>())).Returns(Task.FromResult(expectedValue));
+            _unitOfWork.Setup(x => x.SaveChangesAsync()).Returns(Task.CompletedTask);
+
+            // Act
+            var response = await _achievementService.AddAchievementAsync(model);
+            expectedValue.Id = response.Id;
+
+            // Assert
+            //_fileRepository.Verify(x => x.AddAsync(It.IsAny<File>()), Times.Once);
+            _achievementRepository.Verify(x => x.AddAsync(It.IsAny<Achievement>()), Times.Once);
+            _unitOfWork.Verify(x => x.SaveChangesAsync(), Times.Once);
+            response.Should().BeEquivalentTo(expectedValue);
+        }
+
+        [TestCase]
+        public async Task DeleteAchievementAsync()
+        {
+            //Arrange
+            var achievement = AchievementDumbData.GetRandomEntity();
+
+            _achievementRepository.Setup(x => x.Delete(It.IsAny<Achievement>()));
+            _unitOfWork.Setup(x => x.SaveChangesAsync()).Returns(Task.CompletedTask);
+
+            // Act
+            await _achievementService.DeleteAchievementAsync(achievement.Id);
+
+            // Assert
+            _achievementRepository.Verify(x => x.Delete(It.IsAny<Achievement>()), Times.Once);
+            _unitOfWork.Verify(x => x.SaveChangesAsync(), Times.Once);
+        }
+
+        [TestCase]
+        public async Task GetAchievementByIdAsync()
+        {
+            //Arrange
+            var achievement = AchievementDumbData.GetRandomEntity();
+            var expectedValue = _mapper.Map<ReadAchievementModel>(achievement);
+
+            _achievementRepository.Setup(x => x.GetByIdAsync(It.IsAny<Guid>())).Returns(Task.FromResult(achievement));
+
+            // Act
+            var response = await _achievementService.GetAchievementByIdAsync(achievement.Id);
+
+            // Assert
+            _achievementRepository.Verify(x => x.GetByIdAsync(It.IsAny<Guid>()), Times.Once);
+            response.Should().BeEquivalentTo(expectedValue);
+        }
+
+        [TestCaseSource(typeof(TestCaseSources), nameof(TestCaseSources.ValidUpdateAchievementModels))]
+        public async Task UpdateAchievementAsync_UpdateAchievementModel_ReturnsReadAchievementModel(UpdateAchievementModel model)
```

Comment:
Can you simplify the name of method please.

Reply:
And also in other places, thanks.

Reply:
this is the naming convention 
MethodName_ModelUnderTest_ExpectedBehavior
otherwise it won`t be readable or understandable

Reply:
Ok, now I agree